simple/lint.go

```diff
@@ -1639,3 +1640,47 @@ func (c *Checker) LintRedundantBreak(j *lint.Job) {
 		ast.Inspect(f, fn)
 	}
 }
+
+func (c *Checker) LintStringCopy(j *lint.Job) {
+	addEmptyStr := func(x, y ast.Expr) bool {
+		bl, ok := x.(*ast.BasicLit)
+		if !ok || bl.Kind != token.STRING || bl.Value != `""` {
+			return false
+		}
+		bt, ok := j.Program.Info.TypeOf(y).(*types.Basic)
```

Comment:
`x` must be of underlying type `string`, yeah. Nowadays, all input that we work on has been fully type-checked and would compile.